pyproject.toml

```diff
@@ -30,7 +30,9 @@ opentelemetry-api = { version = "^1.11.1", optional = true }
 opentelemetry-sdk = { version = "^1.11.1", optional = true }
 protobuf = "~4.21"
 python = "^3.7"
+python-dateutil = { version = "^2.8.2", python = "<3.11" }
 types-protobuf = "~3.20"
+types-python-dateutil = { version = "^2.8.19.2", python = "<3.11" }
```

Comment:
I'd be ok with just `# type: ignore` on offending lines instead of adding this dependency

Reply:
The only issue with this is Mypy is complaining on init that its missing stubs, and even `--ignore-missing-import` misses a single error that leads to an exception:

```
(temporalio-py3.10) ubuntu@ip-172-31-27-188:~/projects/sdk-python$ mypy --namespace-packages .
temporalio/converter.py:40: error: Library stubs not installed for "dateutil" (or incompatible with Python 3.10)
temporalio/converter.py:40: note: Hint: "python3 -m pip install types-python-dateutil"
temporalio/converter.py:40: note: (or run "mypy --install-types" to install all missing stub packages)
temporalio/converter.py:40: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

We could solve this by changing the mypy command with the following:
```
mypy --install-types --non-interactive .
```
or add them to `pyproject.toml` instead. What are your thoughts?

Reply:
What happens if you add `# type: ignore` to the end of line 40 of `converter.py` (i.e. the end of the import line)? If that still doesn't help, can you try adding this in `pyproject.toml` right after the `[tool.mypy]` section:

```toml
[[tool.mypy.overrides]]
module = "dateutil"
ignore_missing_imports = true
```

https://github.com/python/mypy/pull/10582 suggests it may help (which is annoying, we already have that set globally).